data/filters/templates/youtube-channel.yaml

```
@@ -11,25 +12,25 @@ tags:
   - youtube
 template: |
   {{#each channels}}
-  www.youtube.com##ytd-browse[page-subtype="home"] ytd-rich-item-renderer:has(a[href$="{{ . }}"])
-  www.youtube.com##ytd-search ytd-video-renderer:has(a[href$="{{ . }}"])
-  www.youtube.com##ytd-search ytd-channel-renderer:has(a[href$="{{ . }}"])
-  m.youtube.com##ytm-browse ytm-rich-item-renderer:has(a[href$="{{ . }}"])
-  m.youtube.com##ytm-search ytm-video-with-context-renderer:has(a[href$="{{ . }}"])
-  m.youtube.com##ytm-search ytm-compact-channel-renderer:has(a[href$="{{ . }}"])
+  www.youtube.com##ytd-browse[page-subtype="home"] a[href$="{{ . }}"]:upward(ytd-rich-item-renderer)
+  www.youtube.com##ytd-search a[href$="{{ . }}"]:upward(ytd-video-renderer)
+  www.youtube.com##ytd-search a[href$="{{ . }}"]:upward(ytd-channel-renderer)
+  m.youtube.com##ytm-browse a[href$="{{ . }}"]:upward(ytm-rich-item-renderer)
+  m.youtube.com##ytm-search a[href$="{{ . }}"]:upward(ytm-video-with-context-renderer)
+  m.youtube.com##ytm-search a[href$="{{ . }}"]:upward(ytm-compact-channel-renderer)
```

With `upward`, you can specify several selectors, and the rule will test each of them while going up the DOM. This will be more efficient than matching on the `href` twice.
```suggestion
  www.youtube.com##ytd-browse[page-subtype="home"] a[href$="{{ . }}"]:upward(ytd-rich-item-renderer)
  www.youtube.com##ytd-search a[href$="{{ . }}"]:upward(ytd-video-renderer,ytd-channel-renderer)
  m.youtube.com##ytm-browse a[href$="{{ . }}"]:upward(ytm-rich-item-renderer)
  m.youtube.com##ytm-search a[href$="{{ . }}"]:upward(ytm-video-with-context-renderer,ytm-compact-channel-renderer)
```